pytrip/vdx.py

```diff
@@ -923,10 +923,7 @@ def get_2d_slice(self, plane, depth):
         for _slice in self.slices:  # TODO: slices must be sorted first, but wouldn't they always be ?
 
             # thanks to previous call to `concat_contour` so there is exactly one contour in each slice
-            contour = np.array(_slice.contours[0].contour)
-            # we need to add first point at the end, without it slice_on_plane won't work properly
-            #   because it operates on closed chains of points
-            contour = np.append(contour, [contour[0]], axis=0)
+            contour = _slice.contours[0].contour
```

Comment:
Was it this place where the time saving occurred ?

Reply:
creating numpy array took almost all the time of calculating intersection

Reply:
nice catch !
Was it due to the fact, that array was relatively small ?

Reply:
creating numpy array was a redundant operation, but using it made C code look cleaner because of numpy C API. 
We changed C code to use operations on classic python arrays, so creating numpy array was not necessary, also instead of adding first point to the end of contour array, we added another for loop body after for loop just for checking last and first point

tl; dr:
creating numpy array wasnt needed and took almost 99 percent of intersecting time